src/dbnode/generated/thrift/rpc.thrift

```diff
@@ -312,6 +314,41 @@ struct HealthResult {
 	2: required string status
 }
 
+enum AggregateQueryType {
+	AGGREGATE_BY_TAG_NAME,
+	AGGREGATE_BY_TAG_NAME_VALUE,
+}
+
+// AggregateQueryRequest comes from our desire to aggregate on incoming data.
+// We currently only support retrieval of facets, but could extend this based on
+// requirements in the future. Currently, the two predominant use-cases are:
+// (1) Given a filter query (optionally), return all known tag keys matching this restriction
+// (2) Given a filter query (optionally), return all know tag key+values matching this restriction
+struct AggregateQueryRequest {
+	1: required Query query
+	2: required i64 rangeStart
+	3: required i64 rangeEnd
+	4: required string nameSpace
+	5: optional i64 limit
+	6: optional AggregateQueryType aggregateQueryType = AggregateQueryType.AGGREGATE_BY_TAG_NAME_VALUE
```

Comment:
I don't think this should be optional; would be better for callers to explicitly state what query type is required

Reply:
Only reason to make it optional is for the case where people just curl this endpoint to see distinct tags. Seems unnecessary to make them also specify this.

That said, I agree - all the stuff calling this endpoint programmatically should totally be specifying this.